account_analytic_salesperson/models/sale_order.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+
+    _inherit = 'sale.order'
+
+    @api.multi
+    def _create_analytic_account(self, prefix=None):
+        """Propagate sale order salesperson through context key"""
+        self = self.with_context(analytic_user_id=self.user_id.id)
```

Comment:
I think this is not correct.

This function is multi and will fail when wanting to access user_id field in multi recordset.

Reply:
Yeah but IMHO that PR is not correct as _create_analytic_account is multi. That one introduces a regression in terms of performance.

Should have been : self.filtered(lambda r: not r.analytic_account_id)._create_analytic_account()

Reply:
We cannot do that here because we need to pass the context for each record.

But anyway, this function is never called on a multi recordset in odoo:
 - https://github.com/odoo/odoo/blob/11.0/addons/sale/models/sale.py#L557
 - https://github.com/odoo/odoo/blob/11.0/addons/sale/models/sale.py#L873
 - https://github.com/odoo/odoo/blob/11.0/addons/sale_expense/models/hr_expense.py#L26
 - https://github.com/odoo/odoo/blob/11.0/addons/sale_timesheet/models/sale_order.py#L202

So it would actually be cleaner to have an `ensure_one` here https://github.com/odoo/odoo/blob/11.0/addons/sale/models/sale.py#L578 but as I see it, there's no real regression in terms of performance.